File: Java/Sorting/BubbleSort.java
import java.util.*;

class BubbleSort{ //time complexity O(n^2)
    static void bubblesort(int arr[]){
        for(int i=0;i<arr.length-1;i++){
            for(int j=0;j<arr.length-1-i;j++){
                if(arr[j] > arr[j+1]){
                int temp = arr[j];
                arr[j]=arr[j+1];
                arr[j+1]=temp;
            }
        }
        }
    }
    static void printArray(int arr[]){
        for(int i=0;i<arr.length;i++){
            System.out.print(arr[i]+ " ");
        }
        System.out.println();
    }
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);

        //size of array
        System.out.print("Enter size of array:");
        int n = sc.nextInt();
        int arr[] = new int[n];

        //array elements
        System.out.print("Enter array elements:");
        for(int i=0;i<n;i++){
            arr[i] = sc.nextInt();
        }
        bubblesort(arr);
        System.out.println("Sorted Elements are ");
        printArray(arr);

    }
}

File: C Program/Sorting Technique/Bubble_Sort.c
#include<stdio.h>

void bubble_sort(int arr[],int n){
    for(int i=0;i<n-1;i++){ //array traversal
        for(int j=0;j<n-i-1;j++){ //array comparison
            if(arr[j]<arr[j+1]){
                int temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
    }
}

void print(int arr[],int n){
    for(int i=0;i<n;i++){
        printf("%d ",arr[i]);
    }
    printf("\n");
}

int main(){
    int n;
    printf("Enter size of an array:");
    scanf("%d",&n);

    int arr[n];
    printf("Enter array elements:");
    for(int i=0;i<n;i++){
        scanf("%d",&arr[i]);
    }

    bubble_sort(arr,n);

    printf("Sorted elements are:");
    print(arr,n);
    return 0;
}